### PR TITLE
TtT chat flow

### PR DESCRIPTION
**Pull request by vitvly**
draft

### CODE REVIEW

**Comment by oskarth** [comment]
@siphiuel might be a good idea to ask for reviewers a la https://github.com/status-im/status-react/blob/develop/doc/decisions/0011-tweak-pr-process.md

File: doc/decisions/0012-tribute-to-talk.md
Comment (oskarth) [inline doc/decisions/0012-tribute-to-talk.md]: > account future migration problems.

If I'm understanding it correctly, is this suggesting contracts can't be upgradable? That seems off to me

Asked @richard-ramos as review, apologize in advance if it's wrong person (forgot who wrote TTT contract)

File: doc/decisions/0012-tribute-to-talk.md
Comment (oskarth) [inline doc/decisions/0012-tribute-to-talk.md]: If we are using IPFS (similar to some other parts of UI), why is there talk about Swarm feeds (really cool, but not quite production-ready/dependable yet)

File: doc/decisions/0012-tribute-to-talk.md
Comment (oskarth) [inline doc/decisions/0012-tribute-to-talk.md]: Kudos for writing ARD with PR

File: doc/decisions/0012-tribute-to-talk.md
Comment (oskarth) [inline doc/decisions/0012-tribute-to-talk.md]: How will this be accepted? Is the plan to bring it up in a core dev call?

**Review by oskarth — CHANGES_REQUESTED** [review]
Didn't review the code, just the ARD. It'd be a good idea to put the code changes under a flag so this PR can be merged sooner rather than later, right now it's been open for a long time.

**Comment by yenda** [comment]
@oskarth it's a draft not a PR, we frequently merge small PRs from it in develop, this is what we use to build and test the big picture

**Comment by oskarth** [comment]
Alright fair enough, the questions in ARD are still relevant though. When is that planned to be merged and communicated?

**Comment by yenda** [comment]
@oskarth good idea to talk about it during dev call. Regarding progress I'm almost done replacing the contract and ipfs by swarm, so I have hstrong hopes the whole thing will be done development wise by Friday.

**Comment by oskarth** [comment]
To be clear, the concern here is more on (seemingly?) relying on a piece of technology that is is still immature. What happens if it stops working? Returns stale reads?

**Comment by yenda** [comment]
@oskarth for different failures:
- data loss: the feed will be checked every time the user logs in and re-uploaded if it is missing
- stale read: as long as we are using a gateway I don't think this can happen?
- gateway down: same as ipfs, spam filter is down while gateway is down

File: src/status_im/ui/screens/chat/views.cljs
Comment (yenda) [inline src/status_im/ui/screens/chat/views.cljs]: remove todo

**Comment by oskarth** [comment]
@yenda @siphiuel Could you explain a bit more on what the purpose of this branch (draft) is? How long is it expected to be kept open? We want to finish up TTT and move to other prios as soon as possible.

**Comment by yenda** [comment]
@oskarth the purpose of this branch is to allow us to test all the code related to tribute to talk. It will probably be closed by the end of the week once the last PR for ttt will be submitted for review. Here is what this branch currently contain that will be in separately reviewed and tested PRs:
- tribute to talk settings #8022 
- fix token transactions #8024 
- fix on-error dispatch in transactions #8021 
- move block contact to separate namespace to avoid circular dependencies #8035
- move ttt document to separate PR to allow more focus debate #8036 

What will be left of this PR after all of the above are merged in develop is going to be the final ttt-chat-flow PR.

**Comment by yenda** [comment]
Update:

I created 4 ttt related issues to fix before release https://github.com/status-im/status-react/issues?q=is%3Aissue+is%3Aopen+label%3Atribute-to-talk

@siphiuel has fixed 2 of them already, the one left are related to how we handle the ugprade to a new version of tribute to talk. The contarct features a way to stop it and we need to ship from the beginning a version of ttt that is aware of that and is deactivated when the contract is stopped with the appropriate UX. We will polish this UX and related issue and submit them as bounties

**Comment by yenda** [comment]
;; account x, valid manifest
satoshi document engage inflict goddess auction rule unfair bid next buddy shy
0x0446e3ea8d84a3783acaefc75502d60f3d5657718c071a72f2cbee7999b1fb8efa2edc41f9d7521a4deb7dc38e7f1bd16d65f55eba450b5ced2c075381324af309

;; account y, valid manifest
plastic prevent toddler soon tuna old door gown slab blood agree pigeon
0x0434b6a979e7e1c7d826bb89f4941b67e7b95f094bb83d8789874dfe52cb00fa866de9bb77d07faab97dd4f1945c608aef6938b7b4d1fdd9af5dd43135f0077a7c

;; invalid manifest
grunt three fragile obtain giggle creek frozen smoke will emerge sheriff sound
0x04c12ef4d3706ae2d8d56e47927f0ce57b29eca19f704125804b6d1b16034f5670b0e6decb3ff010b1e8ea4cb40b5d624eafe50bf06e0ec74299e94faf929241b2
0x6caddd4b2f5ff5375acbec710c870ab382d5085a

Create new account A
Open new chat with account B on mainnet

Screenshot_1556737719

Switch to Ropsten
Open chat with account B

Screenshot_1556737932

Click on pay to chat -> insufficient funds

Screenshot_1556737992

Get some eth and snt
Pay the tribute to B

Screenshot_1556741771
Screenshot_1556741820

**Comment by yenda** [comment]
@churik @errorists see comment above for testing done for tribute payment flow on android

@siphiuel will open a new PR that will fix 2 open bugs, squash all current commits and include testing of tribute payment reception on iphone

**Comment by errorists** [comment]
first of all, thanks for all the work guys! 👏 

_random comment:_ bolded text on Android is not using Inter, have no clue what happened, should I open an issue? 😢 

_and another random comment:_ can we do something about the blurriness of user identicons? can't we request a larger version?

now TtT, I'll test it tomorrow and update this post as I go on but looking at the screenshots:

- [ ] Are you friends? message bubble has wrong left margin, should be the same as regular message

- [ ] Pay to Chat message has a chin, meaning the bottom padding is too large or more likely the text button too much of a height. Let's try to have a 12px distance between the text object and bottom.

- [ ] I've noticed the personalised message and the pay to chat bubble share the same width. It would be nicer if the width of each bubble was calculated individually.

**Review by yenda — APPROVED** [review]

**Comment by vitvly** [comment]
@errorists Thanks for having a look! Fixed the 3 issues you've mentioned. Wrong font/blurriness are still open though. Updated PR is here: https://github.com/status-im/status-react/pull/8129

**Pull request closed without merging**
